File: Java/Labs/Module 2/Lab 1/Q1 - Longest Substring/Main.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s=sc.nextLine();
        int n=s.length();
        int maxLength=0;
        Set<Character> charSet=new HashSet<>();
        int left=0;
        for(int right=0;right<n;right++){
            if(!charSet.contains(s.charAt(right))){
                charSet.add(s.charAt(right));
                maxLength=Math.max(maxLength,right-left+1);
            } else{
                while(charSet.contains(s.charAt(right))){
                    charSet.remove(s.charAt(left));
                    left++;
                }
                charSet.add(s.charAt(right));
            }
        }
        System.out.print(maxLength);
    }
}
                                                                                                                            

File: C++/Labs/Module 2/Lab 1/Q1 - Longest Substring/main.cpp
#include <iostream>
#include <string>
#include <unordered_set>
using namespace std;

int main() {
    string s;
    cin>>s;
    int n=s.length();
    int maxLength=0;
    unordered_set<char> charSet;
    int left=0;
    for(int right=0;right<n;right++){
        if(charSet.count(s[right])==0){
            charSet.insert(s[right]);
            maxLength=max(maxLength,right-left+1);
        } else{
            while(charSet.count(s[right])){
                charSet.erase(s[left]);
                left++;
            }
            charSet.insert(s[right]);
        }
    }
    cout<<maxLength;
    return 0;
}
                                                                                                                            